### PR TITLE
Small updates to dev docs

### LINKED ISSUE #4230: Document how to run screenshot tests
The developer docs don't contain information on how to run the screenshot tests.

### PR DESCRIPTION
**Pull request by birdsarah**
Issue: Fixes #4230 

- Adds some info on screenshot tests
- Surfaces the environment variables
- Small note on heading conventions

### CODE REVIEW

File: sphinx/source/docs/dev_guide/documentation.rst [hunk 1]
```diff
@@ -67,6 +70,27 @@ inserting an implicit "IS" makes a complete sentence. For example::
...
+Headings in pages
+~~~~~~~~~~~~~~~~~
+
+When working on narrative documentation, you may use headings to split up your
+work. Headings are used in the following way
+
+.. code-block:: sh
+  
+    Top level (makes a top-level entry in the side-bar)
+    ====================================================
```
Comment (bryevdv) [inline sphinx/source/docs/dev_guide/documentation.rst]: being pedantic but only because Sphinx would also complain: there is one extra `=` too many

Comment (birdsarah) [inline sphinx/source/docs/dev_guide/documentation.rst]: long live the pedant king :crown: :D

**Comment by birdsarah** [comment]
ping @bryevdv - done!

**Comment by bryevdv** [comment]
LGMT will merge when green

**Pull request merged**